gammapy/scripts/image_pipe.py

```diff
@@ -2,56 +2,180 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 import logging
+import numpy as np
+from astropy.coordinates import Angle
+from astropy.units import Quantity
+from astropy.io import fits
 from ..utils.scripts import get_parser
+from ..background import fill_acceptance_image
+from ..image import SkyMap, SkyMapCollection, disk_correlate
+from ..stats import significance
 
 __all__ = ['ImageAnalysis']
 
 log = logging.getLogger(__name__)
 
 
-def image_pipe_main(args=None):
-    parser = get_parser(ImageAnalysis)
-    parser.add_argument('config_file', type=str,
-                        help='Config file in YAML format')
-    # TODO: add option to dump the default config file
-    args = parser.parse_args(args)
-    analysis = ImageAnalysis.from_yaml(args.config_file)
-    analysis.run()
-
-
 class ImageAnalysis(object):
     """Gammapy 2D image based analysis.
+
+    Parameters
+    ----------
+    empty_image : `~gammapy.image.SkyMap`
```

Comment:
That's a good point. It's not clear to me how flexible the whole `ImageAnalysis` class should be in the end. My suggestion for now would be to only support the single use case, that the user specifies the map properties such as
`nxpix`, `nypix`, `binsz`, `energy_band`, `offset_band` etc. everything else is then computed in the `ImageAnalysis` class step by step.

Later one could think about, that the user can pass a `SkyMapCollection` to the `ImageAnalysis` class and only
if certain maps are missing they are computed in the corresponding methods. What do you think @cdeil? How does it work in the spectral analysis @joleroi?